src/NSubstitute/Core/ThreadLocalContext.cs

```diff
@@ -108,6 +108,24 @@ public IList<IArgumentSpecification> DequeueAllArgumentSpecifications()
         return queue;
     }
 
+    /// <inheritdoc/>
+    public IList<IArgumentSpecification> PeekAllArgumentSpecifications()
+    {
+        var queue = _argumentSpecifications.Value;
+        if (queue == null) { throw new SubstituteInternalException("Argument specification queue is null."); }
```

Comment:
**question:** do you know under what circumstances this occurs? If it is expected can we just return `EmptySpecifications` in that case?

Reply:
tbh it doesn't look like it could ever be null.  I decided to have/keep this in line with enqueue and dequeue methods in case it is something I failed to see

Should we change all 3 (along with enqueue, dequeue) methods to provide a consistent behavior?